e2e/cli-e2e/tests/print-config.e2e.test.ts

```diff
@@ -54,7 +54,70 @@ describe('print-config', () => {
     120000,
   );
 
-  it('should load .ts config file and overwrite it with CLI arguments', async ext => {
+  it.each([
+    // defaults
+    [{}, []],
+    [{ onlyPlugins: 'lighthouse' }, ['lighthouse']],
+    [{ onlyPlugins: ['lighthouse', 'eslint'] }, ['lighthouse', 'eslint']],
+    [{ onlyPlugins: 'lighthouse,eslint' }, ['lighthouse', 'eslint']],
+  ])(
+    'should parse onlyPlugins options correctly',
+    async (options, result) => {
+      const { code, stderr, stdout } = await executeProcess({
+        command: 'code-pushup',
+        args: [
+          'print-config',
+          '--no-verbose',
+          '--no-progress',
+          `--config=${configFile('ts')}`,
+          ...objectToCliArgs(options),
+        ],
+      });
+
+      expect(JSON.parse(stdout)?.onlyPlugins).toEqual(result);
+    },
+    120000,
+  );
+
+  it.each([
+    // defaults
+    [
+      {},
+      {
+        format: PERSIST_FORMAT,
+        filename: 'report',
+      },
+    ],
+    // persist.outputDir
+    [{ 'persist.outputDir': 'my-dir' }, { outputDir: 'my-dir' }],
+    // persist.filename
+    [{ 'persist.filename': 'my-report' }, { filename: 'my-report' }],
+    // persist.format
+    [{ 'persist.format': 'md' }, { format: ['md'] }],
+    [{ 'persist.format': ['md', 'json'] }, { format: ['md', 'json'] }],
+    [{ 'persist.format': 'md,json' }, { format: ['md', 'json'] }],
+  ])(
+    'should parse persist options correctly',
+    async (options, result) => {
+      const { code, stderr, stdout } = await executeProcess({
+        command: 'code-pushup',
+        args: [
+          'print-config',
+          '--verbose',
+          '--no-progress',
+          `--config=${configFile('ts')}`,
+          ...objectToCliArgs(options),
+        ],
+      });
+
+      expect(JSON.parse(stdout)?.persist).toEqual(
+        expect.objectContaining(result),
+      );
+    },
+    120000,
+  );
```

Comment:
E2E tests should be reserved for critical user flows. Testing many different scenarios of argument parsing that most users probably won't encounter isn't cost-effective for E2E tests. Integration (or even unit) tests in `cli` would be a much better fit.

Also, please use include the parameters in the test name when using `it.each` ([`%s`, `%j`, etc.](https://vitest.dev/api/#test-each)), otherwise it creates duplicate test names.

Reply:
Thx will do both!

Reply:
moved